module/modmanager/manager_test.go

```diff
@@ -67,9 +67,15 @@ func TestModManagerFunctions(t *testing.T) {
 	err = mod.startProcess(ctx, parentAddr, logger)
 	test.That(t, err, test.ShouldBeNil)
 
-	err = mod.dial()
+	err = mod.dial(nil)
 	test.That(t, err, test.ShouldBeNil)
 
+	// check that dial can re-use connections.
+	oldConn := mod.conn
+	err = mod.dial(mod.conn)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, mod.conn, test.ShouldResemble, oldConn)
```

Comment:
ShouldResemble doesn't check identity, so it could be a different connection with the same properties and still pass. I think directly using test.ShouldEqual is actually what you want.

Reply:
Good point; fixed.